shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala

```diff
@@ -0,0 +1,71 @@
+package coop.rchain.lmdb
+
+import java.nio.ByteBuffer
+
+import cats.implicits._
+import cats.effect.Sync
+import coop.rchain.shared.Resources.withResource
+import org.lmdbjava.{CursorIterator, Dbi, Env, Txn}
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+final case class LMDBStore[F[_]: Sync](env: Env[ByteBuffer], dbi: Dbi[ByteBuffer]) {
+
+  @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+  // TODO stop throwing exceptions
+  private def withTxn[R](txn: Txn[ByteBuffer])(f: Txn[ByteBuffer] => R): R =
+    try {
+      val ret: R = f(txn)
+      txn.commit()
+      ret
+    } catch {
+      case NonFatal(ex) =>
+        ex.printStackTrace()
+        throw ex
+    } finally {
+      txn.close()
+    }
+
+  def withReadTxnF[R](f: Txn[ByteBuffer] => R): F[R] =
+    Sync[F].delay {
+      withTxn(env.txnRead)(f)
+    }
+
+  def withWriteTxnF[R](f: Txn[ByteBuffer] => R): F[R] =
+    Sync[F].delay {
+      withTxn(env.txnWrite)(f)
+    }
+
+  def get(key: ByteBuffer): F[Option[ByteBuffer]] =
+    withReadTxnF { txn =>
+      dbi.get(txn, key)
+    }.map(v => Option(v))
+
+  @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+  def put(key: ByteBuffer, value: ByteBuffer): F[Unit] =
+    withWriteTxnF { txn =>
+      if (dbi.put(txn, key, value)) {
```

Comment:
I know this code was merely moved but wouldn't :
```
 if (!dbi.put(txn, key, value)) {
        throw new RuntimeException("was not able to put data")
      }
```
be more readable?

Reply:
Good point